backend/dashboard/calculations.py

```diff
@@ -3,211 +3,212 @@
 from rapidfuzz import process
 from dateutil.relativedelta import relativedelta
 
-def _get_closest_match(query: str, choices: dict, score_cutoff: int = 75) -> str:
-    """
-    Returns the best match for query in the keys of choices dict if the score 
-    is above the score_cutoff.
-    """
-    match, score, _ = process.extractOne(query, choices.keys())
-    if score >= score_cutoff:
-        return match
-    return None
-
-def _get_company_env_score(transaction: dict, ESG_scores: dict[str, dict]):
-    """
-    Returns environmental score for a transaction's company IF the company is in the ESG_scores dict.
-    Otherwise, return 0.
-    """
-    company_name = _get_closest_match(transaction['merchant_name'], ESG_scores)
-    if company_name is not None:
-        return ESG_scores[company_name]['environment_score']
-    else:
-        return 0
-
-def _is_green(transaction: dict, ESG_scores: dict[str,dict]):
-    company_env_score = _get_company_env_score(transaction, ESG_scores)
-    return company_env_score > 500
-
-def _company_tier(company_env_score: int) -> int:
-    """
-    Returns the tier of the company based on its environmental score, 
-    worst tier is 4 and best tier is 1.
-    """
-    if company_env_score > 560:
-        return 1
-    elif company_env_score > 520:
-        return 2
-    elif company_env_score > 500:
-        return 3
-    else:
-        return 4
-    
-
-def _get_start_end_dates(current_date: datetime) -> tuple[datetime]:
-    """
-    Helper function, get start and end dates of the month of the current_date.
-    """
-    # Get the first day of the month
-    start_date = current_date.replace(day=1)
-    # Get the last day of the month
-    last_day = monthrange(current_date.year, current_date.month)[1]
-    end_date = current_date.replace(day=last_day)
-    
-    return start_date, end_date
-
-
-
-def _count_green_transactions_in_period(transactions: list[dict], start_date: datetime, end_date: datetime, ESG_scores: dict[str,dict]) -> int:
-    """
-    Helper function that counts green transactions within a specified date range.
-    """
-    return sum(
-        1 for transaction in transactions
-        if start_date <= datetime.strptime(transaction['time_completed'], "%Y-%m-%dT%H:%M:%S.%fZ") <= end_date
-        and _is_green(transaction, ESG_scores)
-    )
-
-
-def _get_unique_companies(transactions: list[dict], ESG_scores: dict[str, dict]):
-    """
-    Return set of top companies shopped at for all time
-    """
-    unique_companies = set()
-
-    for transaction in transactions:
+class Calculations:
+    def _get_closest_match(query: str, choices: dict, score_cutoff: int = 75) -> str:
```

Comment:
I think you need the self parameter in these functions, or need to label them as static

Reply:
Thank you Jennifer for catching that! I fixed it.